### PR TITLE
Improve delete cluster process: unset current-context

### PR DESCRIPTION
**Pull request by kamilhristov**
Unset kubectl currect-context if it is the same as the provided cluster name.

This PR addresses issue #1533

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/1663)

### CODE REVIEW

File: upup/pkg/kutil/kubecfg_builder.go [hunk 1]
```diff
@@ -194,7 +195,14 @@ func (c *KubeconfigBuilder) DeleteKubeConfig() {
...
 	c.execKubectl("config", "unset", fmt.Sprintf("contexts.%s", c.Context))
-	fmt.Printf("Deleted kubectl config for %s\n", c.Context)
+	config, err := clientcmd.LoadFromFile(c.KubeconfigPath)
+	if err != nil {
+		fmt.Printf("Error reading kube config.")
+	}
+	if config.CurrentContext == c.Context {
+		c.execKubectl("config", "unset", "current-context")
+	}
+	fmt.Printf("Deleted kContextubectl config for %s\n", c.Context)
```
Comment (yissachar) [inline upup/pkg/kutil/kubecfg_builder.go]: Typo on `kContextubectl `

File: upup/pkg/kutil/kubecfg_builder.go [hunk 2]
```diff
@@ -194,7 +195,14 @@ func (c *KubeconfigBuilder) DeleteKubeConfig() {
 	c.execKubectl("config", "unset", fmt.Sprintf("users.%s", c.Context))
 	c.execKubectl("config", "unset", fmt.Sprintf("users.%s-basic-auth", c.Context))
 	c.execKubectl("config", "unset", fmt.Sprintf("contexts.%s", c.Context))
-	fmt.Printf("Deleted kubectl config for %s\n", c.Context)
+	config, err := clientcmd.LoadFromFile(c.KubeconfigPath)
+	if err != nil {
+		fmt.Printf("Error reading kube config.")
```
Comment (yissachar) [inline upup/pkg/kutil/kubecfg_builder.go]: Might be better if we add a bit more context, e.g. `Failed to unset kubectl current context: error reading kube config.`. Though this isn't really an error path that should come up ever anyway...

**Comment by yissachar** [comment]
LGTM

**Comment by chrislovecnm** [comment]
@k8s-bot ok to test

**Comment by chrislovecnm** [comment]
@justinsb 1.5.0??  Marking as that, not merging until I get a thumbs up

**Comment by justinsb** [comment]
Thanks @kamilhristov 

We should get this into 1.5.0, but I would like to fix the (unlikely) segfault.  This LGTM other than that.

**Comment by chrislovecnm** [comment]
@justinsb which segfault?

**Comment by justinsb** [comment]
Gah - github was holding my comments in pending :-(

**Comment by kamilhristov** [comment]
@justinsb I created new function DeleteConfig in the kutil package and I am calling it from 'delete cluster' command. What is your opinion on this approach ?

**Comment by justinsb** [comment]
@kamilhristov thanks, but I'm not sure why `DeleteConfig` was needed?    I presume it is because shelling out to kubectl is much slower & requires kubectl on the path, and if so I think this is a good change, but I think it's too big a change for 1.5.0 - although I think we should consider it for an early merge into 1.5.1

**Pull request merged**